nixos/modules/services/web-apps/mealie.nix

```diff
@@ -0,0 +1,108 @@
+{ config, lib, pkgs, ...}:
+let
+  cfg = config.services.mealie;
+  pkg = pkgs.mealie;
+in
+{
+  options.services.mealie = {
+    enable = lib.mkEnableOption "Mealie, a recipe manager and meal planner";
+
+    # TODO allow_signup
+
+    host = lib.mkOption {
+      type = lib.types.str;
+      default = "localhost";
+      description = "Host on which creating the URL on the service";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9000;
+      description = "Port on which to serve the Mealie service";
+    };
+
+    protocol = lib.mkOption {
+      type = lib.types.str;
+      default = "http";
+      description = "Protocol to use to serve the service";
+    };
+
+    logLevel = lib.mkOption {
+      type = lib.types.str;
+      default = "INFO";
+      description = "Log level to use for this service";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.mealie = {
+      description = "Mealie, a self hosted recipe manager and meal planner";
+      after = [
+        "network.target"
+        "network-online.target"
+      ];
+      wantedBy = [
+        "multi-user.target"
+      ];
+
+      environment = {
+        PYTHONPATH = "${pkg.pythonPath}:${pkg}/lib/${pkg.python.libPrefix}/site-packages";
+        STATIC_FILES = "${pkg.frontend}";
+
+        # Data directory defined by systemd service
+        MEALIE_LOG_FILE = "/var/log/mealie/mealie.log";
+        ALEMBIC_CONFIG_FPATH="/var/lib/mealie/alembic.ini";
+
+        # Default configs
+        DEFAULT_GROUP="Home";
+        DEFAULT_EMAIL="changeme@example.com";
+        DEFAULT_PASSWORD="MyPassword";
```

Comment:
Is that the default user's configuration?

If so, I would add an option e.g: `defaultUserConfigFile` to override this, rather than just use bad defaults.

Reply:
Yeah I just copied the launching settings from the sources of `mealie`.
Should I create an option taking an `attrsets` to set these ?
I saw in the documentation that it's possible to use a `.env` file, but it has to be located somewhere specific, and because of all the "paths magic" (links to nix store, frontend not at the same place, etc...) I didn't really bother to use this.
Is that what should be done here ?